rfc/rfc013-vc-irma-proof.md

```diff
@@ -0,0 +1,272 @@
+The `message` is used to connect the signatures and its attributes to a DID. This is the actual text the user will sign.
+The `A` fields identifies which credential is used from the IRMA scheme.
+The `a_disclosed` field contains the actual attribute values. These values are base64 representations of big integers. Big integers are required for the cryptography to work.
+Underneath they contain strings, booleans, integers or enums.
+All the other fields are required to validate the authenticity of the signature. 
+
+The example signature from above will be base64 encoded when used in a Verifiable Credential proof:
+
+Below is a complete example of how an Irma proof can be embedded in a Verifiable Credential:
+
+```json
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://nuts.nl/credentials/v1"
+  ],
+  "id": "did:nuts:t1DVVAs5fmNba8fdKoTSQNtiGcH49vicrkjZW2KRqpv#6f91673b-afa9-4d26-9e0f-00d989943275",
+  "issuanceDate": "2021-03-15T16:34:17.687862+01:00",
+  "issuer": "did:nuts:t1DVVAs5fmNba8fdKoTSQNtiGcH49vicrkjZW2KRqpv",
+  "type": ["VerifiableCredential", "NutsKVKCredential"],
+  "credentialSubject": {
+    "id": "did:nuts:t1DVVAs5fmNba8fdKoTSQNtiGcH49vicrkjZW2KRqpv",
+    "pbdf": {
+      "kvk": {
+        "official": {
+          "legalEntity": "CareBears",
+          "officeAddress": "CarePlace",
+          "kvkNumber": "12345678"
+        }
+      }
+    }
+  },
+  "proof": {
+    "type": "NutsIrmaCredentialProof",
+    "proofValue": "DgYdYMUYHURJLD7xdnWRinqWCEY5u5fK...j915Lt3hMzLHoPiPQ9sSVfRrs1D"
+  }
+}
+```
+
+### 4.2 Verification
+
+Verification of an IRMA signature MUST be done by an [IRMA server](https://irma.app/docs/irma-server/) instance. 
+It's also possible for an implementing party to embed the [IRMA go libraries](https://github.com/privacybydesign/irmago).
+The verification result will indicate if the signature is valid.
+The result will also contain the disclosed attributes as dot-delimited strings.
+All attributes in the `credentialSubject` MUST be present in the result attributes.
+The dot-delimited format of the verification result can be extended as follows:
+
+```text
+A.B.C.D=X
+```
+
+becomes
+
+```json
+{
+  "A": {
+    "B": {
+      "C": {
+        "D": "X"
+      }
+    }
+  }
+}
+```
+
+The last validation is to check if the `message` from the IRMA signature matches the value of `credentialSubject.id`.
+
+### 4.3 IRMA scheme
+
+The IRMA ecosystem has two different schemes: `pbdf` and `irma-demo`. The latter one is not intended for production since all private keys are published with the scheme. Any implementing party MUST ignore all signatures made with that scheme in production.
+
+The contents of the schemes can be viewed on Github: [pbdf](https://github.com/privacybydesign/pbdf-schememanager) and [irma-demo](https://github.com/privacybydesign/irma-demo-schememanager)
+
+### 4.4 Revocation
+
+Although IRMA supports revocation for credentials, this isn't usable for digital signatures.
```

Comment:
But the question is, who's going to issue the revocation when the credential subject equals the issuer? The subject will be a care organization (maybe even an employee's DID in the future?), so if the care organization needs to publish a VC revocation it needs an `assertionMethod` key. But many/most of them won't have one because their software provider it issuing VCs for them?

Also; maybe you can list some real-world scenarios when an IRMA VC credential should be revoked?

Reply:
I added this to §3.3: 

> The proof does not contain a signature created with an `assertionMethod` key from the DID Document.

So the `issuer` in the VC becomes unimportant. The real issuer of the credential is an anonymous person backed by the IRMA issuer.